src/runner/main.sh

```
@@ -197,82 +208,161 @@ test_browser() {
+	if command -v socat &>/dev/null; then
+		socat -T 5 TCP-LISTEN:8930,reuseaddr,fork EXEC:"$STATUS_HANDLER" >/dev/null 2>&1 &
+		STATUS_SERVER_PID=$!
+	elif command -v nc &>/dev/null; then
+		STATUS_FIFO=$(mktemp -u)
+		mkfifo "$STATUS_FIFO"
+		(
+			nc_q=""
+			if nc -h 2>&1 | grep -q -- '-q'; then nc_q="-q 0"; fi
+			while true; do
+				# shellcheck disable=SC2086
+				nc -l 8930 $nc_q <"$STATUS_FIFO" 2>/dev/null | "$STATUS_HANDLER" >"$STATUS_FIFO" || break
```

Shellcheck (0.11.0)

[info] Make sure not to read and write the same file in the same pipeline.

(SC2094)